app/src/shared/modules/user/actions.js

```diff
@@ -239,3 +250,19 @@ export const updatePassword = (passwordUpdate: PasswordUpdate) => (dispatch: Fun
             throw e
         })
 }
+
+export const deleteUserAccount = () => (dispatch: Function) => new Promise((resolve, reject) => {
```

Comment:
Can we go w/o the new `Promise`? Example:
```js
dispatch(deleteUserAccountRequest())
    .then(() => {
        dispatch(deleteUserAccountSuccess())
        dispatch(logout())
    }, (error) => {
        dispatch(deleteUserAccountFailure({
            message: error.message,
        }))
    })
```
Or is it there because of missing API endpoints and faking the actual delete? Ignore my comment if so.

Reply:
I used it here to wait for the action to complete before closing the modal.

Reply:
`services.deleteUserAccount()` is a `Promise` so this
```js
export const deleteUserAccount = () => (dispatch: Function) => {
    dispatch(deleteUserAccountRequest())

    return services.deleteUserAccount()
        .then(() => {
            dispatch(deleteUserAccountSuccess())
            dispatch(logout())
        }, (error) => {
            dispatch(deleteUserAccountFailure({
                message: error.message,
            }))
            throw error // this one is catched later for that `alert(error)`
        })
})
```
should give us pretty much the same effect. Thougts?

Reply:
It's 1 `Promise` less. I'm fine with leaving it as it is. Just throwing another 2 cents on the pile. ;) Cheers!

Reply:
You are right, thanks! I missed the `throw` to pass the error on.